### DISCUSSION TITLE
Typst should emulate the LaTeX page design by default

### DISCUSSION THREAD

**Discussion by blinkerfluessigkeit** [typst/typst#3019] [Ideas]
I've been using Typst for a short while now and so far I'm very impressed.
When it comes to porting old documents to Typst, however, I've had some difficulties. LaTeX is mostly used in Academia and there you have to use certain templates in most cases. As Typst - for example - doesn't set the same whitespace around section/subsection/... headers as LaTeX does, a lot of manual configuration is necessary to be able to actually use Typst productively.

I think it would benefit Typst immensely, if it were to emulate exactly the look of a default LaTeX page, so that people who try it out immediately get the usual look and feel of a LaTeX document and don't get "discouraged". I say this because Academia is very conservative, so a hurdle like this could very well discourage lecturers from using it, if they have to fear that their documents will be rejected for not adhering to certain standards regarding the layout.

An alternative idea to changing the default configuration of Typst would be to offer a template, that emulates the default LaTeX configurations.

**Comment by laurmaedje** [comment]
I think providing LaTeX-like templates is the better choice.

**Reply by blinkerfluessigkeit** [reply]
Yes, I think so too. Would have the advantage of not breaking exisiting documents.

Where do you think we should organize templates? I saw that there is a repo for [templates that are shipped with the webapp](https://github.com/typst/templates) but it says that PRs aren't wanted.

**Reply by Dherse** [reply]
Why do you mean by not breaking documents? You can't trivially translate a latex to typst, so your documents will be broken anyway!?

EDIT: I get it, to not break typst docs my bad.

**Reply by laurmaedje** [reply]
> Where do you think we should organize templates? I saw that there is a repo for [templates that are shipped with the webapp](https://github.com/typst/templates) but it says that PRs aren't wanted.

It is planned to integrate template supports into the [package manager](https://typst.app/docs/packages/). Those would then also show up in the web app's "Start from template" dialogue. For more details, see https://github.com/typst/typst/issues/2432 and [the forge thread on Discord](https://discord.com/channels/1054443721975922748/1176122103355953162).

**Comment by Dherse** [comment]
Why would we want to emulate the terrible presets that latex uses?

> can you make the margin any bigger, I can still fit text on my page

**Reply by blinkerfluessigkeit** [reply]
Constructive criticism is always welcome. Snide, passive aggressive comments are not.

As I said, LaTeX is *the* standard in Academia and certain presets are subsequently cemented into academic writing. Layout presets are amongst those.
I do not wish to discuss whether the layout LaTeX uses is sensible, that is for typographers to discuss. I am confident, however, that the existence of an easy way to emulate LaTeX's default layout means the reduction of the entry barriers for people who have been using LaTeX for a long time.

**Reply by Dherse** [reply]
I wasn't snide towards you, I was being sarcastic and snide with latex's defaults.

Typst does have some templates like the IEEE one, and I don't see any reason to emulate the terrible defaults the latex uses, emulating the templates of publishers? Sure, I would even think it is a necessity. But I think people will get over the default not being ugly.

**Reply by laurmaedje** [reply]
@Dherse Please stay friendly. Even if it comes across that its meant in a sarcastic way, there are more respectful ways to make your point.

**Reply by Dherse** [reply]
@laurmaedje fine you're probably right.

**Reply by astrale-sharp** [reply]
> Why would we want to emulate the terrible presets that latex uses?
This is neither passive agressive nor snide nor is it unfriendly


Can't we be direct in our opinions? As an autistic person, I find that keys to communication are
- be direct
- don't assume intentions

If the question is should we emulate LateX style per default, I'm against it because I also find it to be quite ugly,
that said you can customize typst and there is even a [guide to make it look like LaTex](https://typst.app/docs/guides/guide-for-latex-users/).
If you have criticism of that guide please feel free to discuss it :)

**Reply by mattfbacon** [reply]
It is passive-aggressive and snide because it disregards the justification elaborated in the original post, and it uses a rhetorical question to insult Latex, which is not constructive as blinkerfluessigkeit explained because whether the Latex templates are good-looking is irrelevant to this point. A respectful way of phrasing the meaningful portion of the message would be:

> It might help with familiarity, but I don't like how default Latex documents look (e.g., I think the default margin is too large) and I would prefer not to present that to users by default either.

**Reply by astrale-sharp** [reply]
It's perceived as such (passive-aggressive and snide). Doesn't mean that it is

**Reply by Dherse** [reply]
It is not irrelevant whether the base settings make for a nicer document. That's part of the user experience. Typst doesn't have to emulate latex in every way.

It is relevant for it not to take an hour to set up each and every document to look half decent. It is not irrelevant to improve where latex is... less than ideal. It is not irrelevant to make something better than latex.

This is the same kind of useless comments that people that want shell escape ask for: latex does it and so should typst.

Regarding how i express myself, i was being sarcastic and critical of the presets of latex and i stand by that, i don't have to be tone policed all of the time because somebody doesn't understand sarcasm.
